examples/agents/ibmGraniteBee.ts

```diff
@@ -0,0 +1,125 @@
+import { ChatLLM, ChatLLMOutput } from "bee-agent-framework/llms/chat";
+import { getEnv, parseEnv } from "bee-agent-framework/internals/env";
+import { FrameworkError } from "bee-agent-framework/errors";
+import { TokenMemory } from "bee-agent-framework/memory/tokenMemory";
+import { WatsonXChatLLM } from "bee-agent-framework/adapters/watsonx/chat";
+import { OpenAIChatLLM } from "bee-agent-framework/adapters/openai/chat";
+import { OllamaChatLLM } from "bee-agent-framework/adapters/ollama/chat";
+import { IBMVllmChatLLM } from "bee-agent-framework/adapters/ibm-vllm/chat";
+import { IBMVllmModel } from "bee-agent-framework/adapters/ibm-vllm/chatPreset";
+import { OpenMeteoTool } from "bee-agent-framework/tools/weather/openMeteo";
+import { DuckDuckGoSearchTool } from "bee-agent-framework/tools/search/duckDuckGoSearch";
+import { Ollama } from "ollama";
+import OpenAI from "openai";
+import { z } from "zod";
+import * as process from "node:process";
+import fs from "node:fs";
+
+const Providers = {
+  WATSONX: "watsonx",
+  OLLAMA: "ollama",
+  OPENAI: "openai",
+  IBMVLLM: "ibmvllm",
+  IBMRITS: "ibmrits",
+} as const;
+type Provider = (typeof Providers)[keyof typeof Providers];
+
+const LLMFactories: Record<Provider, () => ChatLLM<ChatLLMOutput>> = {
+  [Providers.OPENAI]: () =>
+    new OpenAIChatLLM({
+      modelId: getEnv("OPENAI_MODEL") || "ibm-granite/granite-3-8b-instruct",
+      parameters: {
+        temperature: 0,
+        max_tokens: 2048,
+      },
+    }),
+  [Providers.OLLAMA]: () =>
+    new OllamaChatLLM({
+      modelId: getEnv("OLLAMA_MODEL") || "granite3-dense:8b",
+      parameters: {
+        temperature: 0,
+        repeat_penalty: 1,
+        num_predict: 2000,
+      },
+      client: new Ollama({
+        host: getEnv("OLLAMA_HOST"),
+      }),
+    }),
+  [Providers.WATSONX]: () =>
+    WatsonXChatLLM.fromPreset(getEnv("WATSONX_MODEL") || "ibm/granite-3-8b-instruct", {
+      apiKey: getEnv("WATSONX_API_KEY"),
+      projectId: getEnv("WATSONX_PROJECT_ID"),
+      region: getEnv("WATSONX_REGION"),
+    }),
+  [Providers.IBMVLLM]: () => IBMVllmChatLLM.fromPreset(IBMVllmModel.GRANITE_INSTRUCT),
+  [Providers.IBMRITS]: () =>
+    new OpenAIChatLLM({
+      client: new OpenAI({
+        baseURL: process.env.IBM_RITS_URL,
+        apiKey: process.env.IBM_RITS_API_KEY,
+        defaultHeaders: {
+          RITS_API_KEY: process.env.IBM_RITS_API_KEY,
+        },
+      }),
+      modelId: getEnv("IBM_RITS_MODEL") || "ibm-granite/granite-3.0-8b-instruct",
+      parameters: {
+        temperature: 0,
+        max_tokens: 2048,
+      },
+    }),
+};
+
+function getChatLLM(provider?: Provider): ChatLLM<ChatLLMOutput> {
+  if (!provider) {
+    provider = parseEnv("LLM_BACKEND", z.nativeEnum(Providers), Providers.OLLAMA);
+  }
+
+  const factory = LLMFactories[provider];
+  if (!factory) {
+    throw new Error(`Provider "${provider}" not found.`);
+  }
+  return factory();
+}
```

Comment:
Could you put everything into one function to make the example more readable?

Reply:
Resolved in 1e88fe1